.circleci/config/jobs/website-docker-image.yml

```diff
@@ -0,0 +1,22 @@
+docker:
+  - image: circleci/buildpack-deps
+shell: /usr/bin/env bash -euo pipefail -c
+steps:
+  - checkout
+  - setup_remote_docker
+  - run:
+      name: Diff package-lock.json
+      command: |
+        echo 'export PACKAGE_LOCK_CHANGED=$(git diff --name-only $(git log --pretty=format:'%h' -n1 HEAD~1)...HEAD | grep -c website/package-lock.json)' >> $BASH_ENV
+  - run:
+      name: Build Docker Image if Necessary
+      command: |
+        if [ "$CIRCLE_BRANCH" = "master" ] && [ $PACKAGE_LOCK_CHANGED -gt 0 ]; then
+          cd website/
+          docker build -t hashicorp/nomad-website:$CIRCLE_SHA1 .
```

Comment:
Rather than relying on dependencies check and/or branch name - I would consider the following approach:

1. Use a stable tag if file change, e.g. `IMAGE_TAG=$(cat website/Dockerfile website/package-lock.json  | sha256sum | awk '{print $1;}')`
2. Only build image if branch is master and `hashicorp/nomad-website:${IMAGE_TAG}` image isn't present 

How does that sound?

Also, how is `hashicorp/nomad-website` image used?  Is it only for local dev purposes?  is it used in deployment pipeline at all?

If this is only for dev purposes and relevant on master - I'd suggest adding a filter so job only runs on master.